lib/load-paths-handler.js

```diff
@@ -27,13 +29,46 @@ class PathLoader {
   }
 
   load (done) {
+    if (this.repo && !this.traverseSymlinkDirectories) {
+      this.loadFromGit().then(done)
+
+      return
+    }
+
     this.loadPath(this.rootPath, true, () => {
       this.flushPaths()
       if (this.repo != null) this.repo.destroy()
       done()
     })
   }
 
+  async loadFromGit () {
+    return new Promise((resolve) => {
+      const args = ['ls-files', '--cached', '--exclude-standard', '--others', '-z']
+
+      for (let ignoredName of this.ignoredNames) {
+        args.push('--exclude', ignoredName.pattern)
+      }
+
+      let output = ''
+
+      // TODO: do this via a call to GitRepository (needs to be implemented).
```

Comment:
We have quite a bit of infrastructure within atom/github to interact with external git processes - with queueing, caching, and a sidecar worker process to fix a sporadic performance issue with `child_process` calls... see atom/github#386 and atom/github#688 for some background on _that_ particular mess 😆 . We also bundle a git executable, so we don't need to rely on a pre-existing user installation. Rather than calling `GitRepository`, we should publish a [package service](https://flight-manual.atom.io/behind-atom/sections/interacting-with-other-packages-via-services/) that allows other packages like this one to make git calls on demand. See atom/github#1089 for some thoughts on this - this might be a good motivating case :smile: